pkg/query-service/app/preferences/model.go

```diff
@@ -0,0 +1,510 @@
+	tableSchema := `
+	PRAGMA foreign_keys = ON;
+	CREATE TABLE IF NOT EXISTS user_preference(
+		preference_id TEXT NOT NULL,
+		preference_value TEXT,
+		user_id TEXT NOT NULL,
+		PRIMARY KEY (preference_id,user_id),
+		FOREIGN KEY (user_id)
+			REFERENCES users(id)
+			ON UPDATE CASCADE
+			ON DELETE CASCADE
+	);`
+
+	_, err = db.Exec(tableSchema)
+	if err != nil {
+		return fmt.Errorf("error in creating user_preference table: %s", err.Error())
+	}
+
+	// create the org preference table
+	tableSchema = `
+	PRAGMA foreign_keys = ON;
+	CREATE TABLE IF NOT EXISTS org_preference(
+		preference_id TEXT NOT NULL,
+		preference_value TEXT,
+		org_id TEXT NOT NULL,
+		PRIMARY KEY (preference_id,org_id),
+		FOREIGN KEY (org_id)
+			REFERENCES organizations(id)
+			ON UPDATE CASCADE
+			ON DELETE CASCADE
+	);`
+
+	_, err = db.Exec(tableSchema)
+	if err != nil {
+		return fmt.Errorf("error in creating org_preference table: %s", err.Error())
+	}
+
+	preferenceFromFile, fileErr := fs.ReadFile(os.DirFS("../../pkg/query-service/app/preferences"), "preference.json")
+
+	if fileErr != nil {
+		return fmt.Errorf("error in reading preferences from file : %s", fileErr.Error())
+	}
+
+	if unmarshalErr := json.Unmarshal(preferenceFromFile, &preferences); unmarshalErr != nil {
+		return fmt.Errorf("error in unmarshalling preferences from file : %s", unmarshalErr.Error())
+	}
+
+	preferenceMap = map[string]Preference{}
+	for _, preference := range preferences {
+		_, seen := preferenceMap[preference.Key]
+		if seen {
+			return fmt.Errorf("duplicate preference key in the preferences: %s", preference.Key)
+		}
+		preferenceMap[preference.Key] = preference
+	}
+
+	return nil
+}
+
+// org preference functions
+func GetOrgPreference(ctx context.Context, preferenceId string, orgId string) (*PreferenceKV, *model.ApiError) {
+	// check if the preference key exists or not
+	preference, seen := preferenceMap[preferenceId]
+	if !seen {
+		return nil, &model.ApiError{Typ: model.ErrorBadData, Err: fmt.Errorf("no such preferenceId exists: %s", preferenceId)}
+	}
+
+	// check if the preference is enabled for org scope or not
+	isPreferenceEnabled := preference.IsEnabledForScope(OrgAllowedScope)
+	if !isPreferenceEnabled {
+		return nil, &model.ApiError{Typ: model.ErrorForbidden, Err: fmt.Errorf("preference is not enabled at org scope: %s", preferenceId)}
+	}
+
+	// fetch the value from the database
+	var orgPreference PreferenceKV
+	query := `SELECT preference_id , preference_value FROM org_preference WHERE preference_id=$1 AND org_id=$2;`
+	err := db.Get(&orgPreference, query, preferenceId, orgId)
+
+	// if the value doesn't exist in db then return the default value
+	if err == sql.ErrNoRows {
+		return &PreferenceKV{
+			PreferenceId:    preferenceId,
+			PreferenceValue: preference.DefaultValue,
+		}, nil
+	} else if err != nil {
+		return nil, &model.ApiError{Typ: model.ErrorExec, Err: fmt.Errorf("error in fetching the org preference: %s", err.Error())}
+	}
```

Comment:
Always try doing the nil check first! It cultivates a good habit to check with pointers and is a little more idiomatic

Reply:
done